TickSpec.Tests/FeatureParserTest.fs

```diff
@@ -16,20 +16,30 @@ let private verifyParsing (lines: string[]) (expected: FeatureSource) =
 let private verifyLineParsing (lines: string[]) (expected: LineType list) =
     let lineParsed =
         lines
-        |> Seq.map (fun line ->
+        |> Seq.mapi (fun lineNumber line -> (lineNumber + 1, line))
+        |> Seq.map (fun (lineNumber, line) ->
             let i = line.IndexOf("#")
-            if i = -1 then line
-            else line.Substring(0, i)
+            if i = -1 then lineNumber, line
+            else lineNumber, line.Substring(0, i)
         )
-        |> Seq.filter (fun line -> line.Trim().Length > 0)
-        |> Seq.scan (fun (_, lastLine) line ->
-            let parsed = parseLine (lastLine, line)
+        |> Seq.scan (fun prevLine (lineNumber, lineContent) ->
+            let lastParsedLine =
+                match prevLine with
+                | Line (_, _, prevLineType) -> prevLineType
+                | IgnoredLine prevLineType -> prevLineType
+
+            let parsed = parseLine (lastParsedLine, lineContent)
             match parsed with
-            | Some line -> (lastLine, line)
+            | Some line -> (lineNumber, lineContent, line) |> Line
+            | None when lineContent.Trim().Length = 0 -> lastParsedLine |> IgnoredLine
             | None ->
-                let e = expectingLine lastLine
-                Exception(e) |> raise) (FileStart, FileStart)
-        |> Seq.map (fun (_, line) -> line)
+                let e = expectingLine lastParsedLine
+                Exception(e) |> raise) (Line (0, "", FileStart))
+        |> Seq.choose (function
```

Comment:
```suggestion
        |> Seq.filter (function IgnoredLine -> false | Line -> true)
```

Reply:
But on reflection you could collapse the `filter >> map` into a `choose`:
` |> Seq.choose (function IgnoredLine -> None | Line (_, _, line) -> Some line`  ?

Reply:
I used the suggestion - you are right, there is no point to have `choose` followed by `map`.